dev/setup-k3d

```diff
@@ -7,6 +7,14 @@ docker_mirror=${docker_mirror-}
 name=${1-upstream}
 i=${2-0}
 
+if [ ! -z METRICS_CONTROLLER_PORT ]; then
```

Comment:
I'm not too happy vs `setup-k3ds` vs. `setup-k3d`. In the beginning I used `setup-k3d` to set individual clusters, but both scripts diverged and now integrationtests only work with `setup-k3d` 🤷 

https://github.com/rancher/fleet/pull/3156/files#r1892583417